platform/core/src/feature_file_system.rs

```diff
@@ -0,0 +1,495 @@
+use crate::{core_failure::CoreFailure, io::file_system::FileSystem};
+use url::Url;
+use uuid::Uuid;
+
+trait FeatureFSConfigTrait {
+    fn features_path(&self) -> Result<String, String>;
+    fn feature_name(&self) -> Result<String, String>;
+}
+struct FeatureFSConfig {}
+
+impl FeatureFSConfigTrait for FeatureFSConfig {
+    fn features_path(&self) -> Result<String, String> {
+        // TODO: get the path from the platform
+        Ok(env!("CARGO_MANIFEST_DIR").to_string() + "/TestFeatures/")
+    }
+
+    fn feature_name(&self) -> Result<String, String> {
+        // TODO: get the feature name from the platform
+        Ok("Test".to_string())
+    }
+}
+
+fn feature_files_path(config: &impl FeatureFSConfigTrait) -> Result<String, CoreFailure> {
+    let features_path = config
+        .features_path()
+        .map_err(CoreFailure::failed_to_create_feature_files_path)?;
+    let feature_name = config
+        .feature_name()
+        .map_err(CoreFailure::failed_to_create_feature_files_path)?;
+    Ok(features_path + "/" + &feature_name)
+}
+
+type ResourceUrl = String;
+#[allow(dead_code)]
+type ResourceId = String;
+
+#[allow(dead_code)]
+fn resource_url_from_id(id: &ResourceId) -> ResourceUrl {
+    let res_prefix = "polypod://FeatureFiles".to_string();
+    res_prefix + "/" + id
+}
+
+#[allow(dead_code)]
+fn fs_path_from_id(
```

Comment:
If all we're doing with the URI (not URL) is adding and removing the prefix, what purpose does it serve, really?